nowcasting_dataset/manager.py

```diff
@@ -0,0 +1,384 @@
+"""Manager class."""
+
+import logging
+from concurrent import futures
+from pathlib import Path
+from typing import Optional, Union
+
+import numpy as np
+import pandas as pd
+
+# nowcasting_dataset imports
+import nowcasting_dataset.time as nd_time
+import nowcasting_dataset.utils as nd_utils
+from nowcasting_dataset import config
+from nowcasting_dataset.data_sources import ALL_DATA_SOURCE_NAMES, MAP_DATA_SOURCE_NAME_TO_CLASS
+from nowcasting_dataset.dataset.split import split
+from nowcasting_dataset.filesystem import utils as nd_fs_utils
+
+logger = logging.getLogger(__name__)
+
+
+SPATIAL_AND_TEMPORAL_LOCATIONS_OF_EACH_EXAMPLE_FILENAME = (
+    "spatial_and_temporal_locations_of_each_example.csv"
+)
+
+
+class Manager:
+    """The Manager initialises and manage a dict of DataSource objects.
+
+    Attrs:
+      config: Configuration object.
+      data_sources: dict[str, DataSource]
+      data_source_which_defines_geospatial_locations: DataSource: The DataSource used to compute the
+        geospatial locations of each example.
+      save_batches_locally_and_upload: bool: Set to True by `load_yaml_configuration()` if
+        `config.process.upload_every_n_batches > 0`.
+      local_temp_path: Path: `config.process.local_temp_path` with `~` expanded.
+    """
+
+    def __init__(self) -> None:  # noqa: D107
+        self.config = None
+        self.data_sources = {}
+        self.data_source_which_defines_geospatial_locations = None
+
+    def load_yaml_configuration(self, filename: str) -> None:
+        """Load YAML config from `filename`."""
+        logger.debug(f"Loading YAML configuration file {filename}")
+        self.config = config.load_yaml_configuration(filename)
+        self.config = config.set_git_commit(self.config)
+        self.save_batches_locally_and_upload = self.config.process.upload_every_n_batches > 0
+
+        # TODO: This should probably be done in the Pydantic model.
```

Comment:
maybe, maybe not. the data will be the same if a different temp path is used

Reply:
but I agree it may be useful to define it in the config - (low priority)

Reply:
Yeah, I agree, this is low priority.  I've added a new issue to discuss this: #320